lib/datadog/core/remote/client/capabilities.rb

```diff
@@ -23,7 +23,7 @@ def initialize(settings)
           private
 
           def register(settings)
-            if settings.appsec.enabled
+            if settings.respond_to?(:appsec) && settings.appsec.enabled
```

Comment:
To be honest, this is why I'm not a big fan of patching behavior into core classes, especially conditionally; it makes is a bit hard to reason about what's there or what's not and only really gives the illusion of things being separate.

Reply:
Agree 👍